File: src/main/java/com/gitlab/daring/image/common/BaseComponent.java
package com.gitlab.daring.image.common;

import com.typesafe.config.Config;

import static com.gitlab.daring.image.config.ConfigUtils.defaultConfig;

public abstract class BaseComponent {

    public final Config config;

    protected BaseComponent(Config config) {
        this.config = config;
    }

    protected BaseComponent(String configPath) {
        this(defaultConfig(configPath));
    }

    protected Config getConfig(String path) {
        return config.getConfig(path);
    }

}

File: src/main/kotlin/com/gitlab/daring/image/common/BaseComponent.kt
package com.gitlab.daring.image.common

import com.gitlab.daring.image.config.ConfigUtils.defaultConfig
import com.typesafe.config.Config

abstract class BaseComponent(@JvmField val config: Config) {

    constructor(configPath: String) : this(defaultConfig(configPath))

    fun getConfig(path: String): Config {
        return config.getConfig(path)
    }

}